WordPress/gutenberg
# Archives: Make the label above the dropdown editable

## What?
Adds an editable label attribute to the Archives block that is displayed when "display as a dropdown" is selected.

## Why?
https://github.com/WordPress/gutenberg/issues/57528
Currently the dropdown is not editable.

## How?
This initial commit puts the label textControl in the sidebar which is suboptimal. It should be directly editable. 

## Testing Instructions
1. Open a post or page.
2. Insert an archives block.
3. Set to display as a dropdown and show label
4. Set a custom label. 
5. Note that toggling the display as dropdown or show labels shows/hides the textControl for the label correctly.
6. save post and confirm label is shown on the front end and the editor.

### Testing Instructions for Keyboard
Similar to cursor, but confirm that tab controls correctly include the new textControl to set the label. The tab control works correctly when the textControl is removed when the display as dropdown is unchecked.

<img width="980" alt="Screenshot 2024-01-05 at 2 12 47 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/5db3f79a-aaaf-4362-a4e3-80424304ace5">
